File: ruby/CP/HackerRank/insertionsort1.rb
def insertion_sort(arr, len)
    e = arr[-1]
    y = len - 1

    while (y>0) && (arr[y-1] > e)
        arr[y] = arr[y-1]
        y -= 1
        puts arr.join(" ")
    end

    arr[y] = e
    puts arr.join(" ")
end


len = gets.chomp.to_i
arr = gets.chomp.split.map { |x| x.to_i }
insertion_sort(arr, len)

File: Competive Programming/Python/HackerRank/insertionsort1.py
def insertion_sort(arr, length):
    e = arr[-1]
    x =  length - 1

    while (x>0) and (arr[x-1] > e):
        arr[x] = arr[x-1]
        x -= 1
        print(*arr)
    arr[x] = e
    print(*arr)


length = int(input())
arr = list(map(int, input().split()))
insertion_sort(arr, length)

